### PR TITLE
Add `WoodburyEstimator` for high dimensionality

### PR DESCRIPTION
**Pull request by timholy**
When the covariance matrix is too large to handle in full, this provides the option to model it as `Σ = σ²I + U * Λ * U'` for some low-rank `U` and diagonal `Λ`.

### CODE REVIEW

**Comment by timholy** [comment]
Documenter failure appears to be due to versioning: it ran with Julia 1.4. The more serious issue is that I didn't want to keep supporting 1.0 in WoodburyMatrices, so v1 (which is required for this PR) requires a minimum of Julia 1.6. How do you feel about bumping the minimum version here?

**Comment by mateuszbaran** [comment]
Thanks, I will check it out tomorrow :+1: . I think we can bump minimum Julia version to 1.6. We haven't done it yet because nothing was broken on Julia 1.0 but now most of the ecosystem has moved on.

File: src/loss.jl [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+| 6          | `sqrt(A) \\ B / sqrt(A) - I` | |
+| 7          | `log(sqrt(A) \\ B / sqrt(A))` | Not supported |
+
+See also [`StatLoss`](@ref).
+
+Reference:
+    Donoho, D.L., Gavish, M. and Johnstone, I.M., 2018.
+    Optimal shrinkage of eigenvalues in the spiked covariance model. Annals of statistics, 46(4), p.1742.
+"""
+struct NormLoss <: LossFunction
```
Comment (mateuszbaran) [inline src/loss.jl:40]: Please add these docstrings to https://github.com/mateuszbaran/CovarianceEstimation.jl/blob/master/docs/src/lib/public.md#public-interface

**Review by mateuszbaran — COMMENTED** [review]
Everything else looks fine :+1:

**Comment by mateuszbaran** [comment]
I've fixed some issues with documentation, could you merge `master` branch after addressing my review?

**Comment by timholy** [comment]
Merging against master will be fine.

Sorry for the delay here, I've found what appear to be discrepancies between the figures and the formulas in the paper. They provide Matlab code but it's GPL-licensed, so I can't look at it to try to resolve the discrepancy. I'm contacting the authors to see if I can get a resolution.

**Comment by mateuszbaran** [comment]
Don't worry about the delay, it's great that you work on new contributions. I hope the authors will be helpful.

**Comment by timholy** [comment]
The authors (and a postdoc) were extremely helpful, and the error turned out to be my own!

I think this may be ready. Recent commits expanded the docs considerably, so it could be worth another look.

**Review by mateuszbaran — APPROVED** [review]
Thank you, the new estimator looks really well.

**Pull request merged**